hack/tools/go.mod

```diff
@@ -7,69 +7,77 @@ godebug default=go1.24
 require (
 	github.com/aojea/sloppy-netparser v0.0.0-20210819225411-1b3bd8b3b975
 	github.com/cespare/prettybench v0.0.0-20150116022406-03b8cfe5406c
-	github.com/golangci/golangci-lint v1.64.5
+	github.com/golangci/golangci-lint/v2 v2.1.5
```

Comment:
starting to think about moving tools into seperate modules, so they behave closer to if you just `go install`ed them and don't conflate dep versions, there are some tradeoffs (sharing deps improves build caching for them marginally), it's hard to tell if any of these impact other tools

Reply:
if you want to do that, I recommend reading the section `go tool` of the golangci-lint doc https://golangci-lint.run/welcome/install/#install-from-sources

Reply:
Right, currently we have this issue from the other tools:
> When using the "tools pattern" or tool command/directives, the dependencies of a tool can modify the dependencies of another tool or your project. The resulting binary was not tested and is not guaranteed to work.

Tracking it in a shared tools module has risks, I think we should split it out.
Then the only question is if we use a specific module or avoid a module entirely.
Using a module means we get GOSHASUM and other visibility, at the risk of breaking replace statements.

Reply:
We prefer not to take opaque binary dependencies generally, and it's also more complex to do portably across OS-es. It is somewhat less opaque to track this way, and I see shasums breaking *as a feature* that indicates some unknown change was made that hasn't been looked at. The GOSHASUMS enable TOFU in a way that is much simpler and re-usable than binaries.

Reply:
I suggested that you read the section "`go tool` usage recommendations`.
You have to click on the triangle.

Reply:
Ahh! it wasn't obvious that this expanded, and the lines above at the linked heading talk about `go tool`

Might be worth tweaking the styling / layout.

Reply:
It is intentionally not a heading because we strongly don't recommend that.
But I prefer to provide the right way to do it instead of letting users in the dark.
So this is a compromise.

Reply:
Installing golangci-lint through a separate module seems worthwhile. I've added one more commit doing that:

```
golangci-lint: move into hack/tools/golangci-lint

The advantage is that it separates the dependencies of the different tools
better. golangci-lint in particular has many dependencies and is sometimes
sensitive to the exact version being used. This way, "go get" bumps up
dependencies exactly as defined by the upstream golangci-lint module.

It's not quite self-contained because logcheck as a Go plugin for golangci-lint
must be built from the same dependencies. But it only adds one and does not
change any of the others.

While at it, the Go 1.24 "tools" directive gets used instead of the traditional
tools.go approach.
```

Reply:
As explained inside the doc, you don't need a "full" new module, you can use only a dedicated module file.
I don't say this is a better approach, this is just to be sure this approach has been evaluated.